core/src/main/java/com/graphhopper/routing/util/spatialrules/SpatialRule.java

```diff
@@ -29,12 +29,17 @@
  */
 public interface SpatialRule {
 
-    double getMaxSpeed(ReaderWay readerWay, String transportationMode);
+    String MAXSPEED_KEY = "max_speed";
+    String ACCESS_KEY = "access";
 
-    AccessValue isAccessible(ReaderWay readerWay, String transportationMode);
+    Object getObject(String key, ReaderWay readerWay, String transportationMode, Object _default);
+
+    <T> T get(String key, ReaderWay readerWay, String transportationMode, T _default);
 
     List<Polygon> getBorders();
+
     SpatialRule setBorders(List<Polygon> borders);
+
     SpatialRule addBorder(Polygon polygon);
 
     String getCountryIsoA3Name();
```

Comment:
I think getCountryIsoA3Name is also too specific e.g. imagine the use case from phil where you have circles around a city to define different areas for the speeds. Maybe we keep everything more specific for now and switch later to a more generic version? The problem I see is that although getObject is already very generic, the parameters could be limited ... or should we remove transportationMode e.g. I can image that sometimes this makes no sense. Maybe the best bet would be `getObject(key, ConfigMap, _default)` and we provide a fast 'converter' from ReaderWay to ConfigMap ... hmmh ... not clear. Similar problem like we have for #472